### DISCUSSION TITLE
Default noise behavior for seed

### DISCUSSION THREAD

**Discussion by beckynevin** [deepskies/DeepBench#74] [Q&A]
I'm working on the plotting tools for the pendulum and I realized that the noise function defaults to the random seed 42 each time we run it. This means that when we run it several times in a row, it will return identical output. Is this the default behavior we want or do we want it to use a new random seed each time and only use the seed=42 when specified?

I'm thinking about functionality in inference down the line where we might want to re-run multiple noise iterations and have it by default choose a new random seed. I'm also wondering how we want it to choose the random seed, because giving it a range of integers also seems restrictive.

I've attached a plot of what it would look like to by default draw the noise from a new random seed each time.
<img width="572" alt="image" src="https://github.com/deepskies/DeepBench/assets/26393168/81f8be56-2962-40ba-8b9d-598a6c382369">

**Comment by voetberg** [comment]
Maybe a good solution would be to automatically set a new seed if it wasn't set already and pass that back to be included in the returned object? 

@AeRabelais Is the seed set in the catalogue init so should each class be handling them uniquely?

**Comment by beckynevin** [comment]
Okay for now my solution is to change the default behavior to be seed: int = None, so there still is an option to specify a seed but if they don't then it's random.

**Comment by beckynevin** [comment]
I'm starting another branch to work on the logging that saves the seed. @voetberg and @bnord , I'm wondering if we also want to save any additional info in the logging. I'm concerned that I'm mixing this train of thought up with the config file that will be generated after every run (the h5?). Here are some questions I have:

- How will we associate a line of this log file with a specific run? 
- Should we also be saving the time array and the output positions also? (this seems like its already taken care of in the h5)
- How often should we delete the log file or should the default be to create a new log file with a new run number each time? (this seems like we're getting into config file territory)
